### ISSUE TITLE
Cannot login in

### ISSUE THREAD

**Issue by chrisstayton** [wpo365/wpo365-login#40]
Using the FREE version of the plugin; after installation, per tutorial video, I go to Users > Add New to create a new user. I then add a user from my O365 tenant. I do not select a password for them. I create the account. I completely log out of the admin account and use incognito mode in chrome to log in using the O365 user. When trying to log in on the front-end of the website using the O365 user, I get a "Login Failed! You entered an invalid username or password." When trying to log in on the back-end of the website using O365 user, I get "ERROR: Invalid username." What could be the issue? Please help!

**Comment by wpo365** [comment]
Hi @chrisstayton It seems to me that you describe errors that you'd normally see when you sign in manually using the wp login form by entering your username and password. With the plugin, however, you should, as soon as you navigate to the wordpress back end (or back- and front end depending on the selected authentication scenario) you should automatically be redirected to Microsoft to authenticate. Can you confirm that this isn't happening but that you instead try to log on manually?

Ps, feel free continuing contacting me through the website's contact form http://www.wpo365.com/contact/

**Comment by chrisstayton** [comment]
Correct. I do NOT get redirected to any Microsoft login page. It only goes to the normal wp-login.php

**Comment by wpo365** [comment]
@chrisstayton What does happen when you click the "Test authentication" button on the Single Sign-on wizard tab? You should see that a popup opens (if not, disable the popup blocker) and then it should close if it's all set up correctly. If not, you'll probably see the error ...

**Comment by chrisstayton** [comment]
A pop up opens perfectly and "Authentication tested successfully" message appears by "Test Authentication" button.

**Comment by wpo365** [comment]
Have you by any change added / and /wp-admin/ to the Pages Blacklist page? And what is the selected authentication scenario?

**Comment by chrisstayton** [comment]
Yes, I have. both http:// & https://

Authentication scenario is selected to be on "internet"

**Comment by wpo365** [comment]
You mean that you have added http(s)://www.yourwebsite.com/wp-admin/ to the Pages Blacklist? That would basically disable the Plugin for the wp Admin area. So you should delete this. With internet as the selected scenario the Plugin would skip trying to authenticate the user when you access any page or post.

**Comment by wpo365** [comment]
@chrisstayton I haven't heard back from you so I assume you've been able to fix it?

**Comment by chrisstayton** [comment]
No sir. Sorry was out of the country.

**Comment by chrisstayton** [comment]
So, maybe I am not explaining the situation completely. 

I have pages that I need to keep restricted to the public. I am using WP-Members to block those pages. The WPO365 login fails when using the WP-Members plugin and when the plugin is disabled. I removed the http(s)://www.intercotradingco.com/wp-admin/ from the blacklist too, the error still occurs.

**Comment by wpo365** [comment]
I don't know about the WP-Member plugin. However, you don't need to add all public pages to the Pages Blacklist either when you have selected Internet as the Authentication Scenario. With this scenario. all front-end pages are basically generally available to all anonymous users. If you choose Intranet as the Authentication Scenario, all front-end pages become private apart for the pages in the blacklist. 
You write that the plugin fails. Can you tell more about this error?

Ps, better delete the above image and update any secrets (or even better create a new application registration with a new ID and secrets)!

**Comment by chrisstayton** [comment]
Okay. So my best option to accomplish my goal is to...

1. Use the Intranet authentication scenario?
2. Add pages, I want the public to access, to the "Pages blacklist"?

After I do that, will the page I want to be secure (without using wp-members) be redirected to the O365 login page?

Yes, I will definitely now create a new app with new id and secrets.

**Comment by wpo365** [comment]
Basically yes, but with that scenario you can not have your landing page public - there is no way to add "/" to the pages blacklist without making all pages again publicly available ... I'm afraid it's a bit of a dead end street ... At the moment I think the "mixed" scenario is not implementable with solely this plugin ... I know of others who used plugins to make front-end pages private so that users must be logged in to the website first before being able to access those and add a link to sign in with Microsoft to their login page (using the premium version's short code) https://www.wpo365.com/authentication-shortcode/ This way you can offer a public website, have some pages requiring login. redirecting the user to the login form and there they then will be able to click "Sign in with Microsoft" which basically invokes the authentication offered by the plugin manually.

**Comment by chrisstayton** [comment]
Okay. So here is what I did. 

I created a subdomain (itcep.intercotradingco.com). I want the entire site to be used as an "Intranet". I installed the plugin and everything tested fine regarding authentication. But when I actually go to the website login using the O365 credentials, it tries to redirect a few moments then it displays an error 

> This page isn’t working itcep.intercotradingco.com didn’t send any data.
> ERR_EMPTY_RESPONSE

I think it is some sort of loop. How do I fix it? 

Ive included a screenshot of the settings
![wpo365 wizard itc employee portal wordpress](https://user-images.githubusercontent.com/46897592/52145312-40d7ad80-2626-11e9-9547-c0d50eb67ef2.png)

**Comment by wpo365** [comment]
Let's continue per direct email - it's easier and you don't need to "hide" all details. Please contact me direct at info at wpo365.com. In the mean time, what happens if you click Test Authentication? Normally, when the application is looping, I recommend that - when you use Chrome browser - start Developer Tools (F12), activate the Network Tab, click Preseve log (and disable cache) and then look at the log ... You can send me a screenshot and maybe I can see something's wrong.

**Comment by chrisstayton** [comment]
When I click on Test Authentication, a pop-up window appears and closes. I get an “Authentication tested successfully”. I've attached a PDF of screenshots I took while looking at the log in Developer mode in Chrome.

Please help with this issue. Thank you so much for your time and efforts.

Christian Stayton
[Interco Trading Co.]<http://www.intercotradingco.com/>
10 Fox Industrial Park Bldg 3 • Madison, IL 62060
(618) 798-9500 Main
(618) 501-0223 Direct
(618) 798-9501 Fax
(319) 529-8178 Cell
www.intercotradingco.com<http://www.intercotradingco.com/>
cstayton@intercotradingco.com<mailto:cstayton@intercotradingco.com>
[cid:image002.jpg@01D469FA.4AFBBF60]
[cid:image003.jpg@01D469FA.4AFBBF60]
[linkedin64]<https://www.linkedin.com/company/intercotrading>[facebook64]<https://www.facebook.com/IntercoTradingCo>[twitter64]<https://twitter.com/intercotrading>[Instagram64]<https://www.instagram.com/intercotradingco>[cid:image008.png@01D469FA.4AFBBF60]<https://www.youtube.com/channel/UCdxZLAmDGPDddnD9mLlRoeg>

**Comment by chrisstayton** [comment]
Okay. I sent you an email. Please resond as soon as possible. Thank you for all of your help!

**Comment by Chuvantinh** [comment]
Hi ,

did you find any solutions for this problem ? I have got also same problem. After success login with microsoft 365 I have an account in wordpress databases. Then I try to use form login of wordpress but I can not sign in. 

is that some thing wrong in config or is this not possible  ? 

thanks

**Comment by wpo365** [comment]
@Chuvantinh Please use either the [WordPress Support Forum](https://wordpress.org/support/plugin/wpo365-login/) or [contact us directly via the website](https://www.wpo365.com/contact/).
